src/field_control_avatar.c

```diff
@@ -1053,3 +1079,113 @@ int SetCableClubWarp(void)
     SetupWarp(&gMapHeader, GetWarpEventAtMapPosition(&gMapHeader, &position), &position);
     return 0;
 }
+
+static bool32 TrySetUpWalkIntoSignpostScript(struct MapPosition *position, u32 metatileBehavior, u32 playerDirection)
+{
+    const u8 *script;
+
+    if ((JOY_HELD(DPAD_LEFT | DPAD_RIGHT)) || (playerDirection != DIR_NORTH))
+        return FALSE;
+
+    switch (GetFacingSignpostType(metatileBehavior, playerDirection))
+    {
+    case MB_POKEMON_CENTER_SIGN:
+        SetUpWalkIntoSignScript(Common_EventScript_ShowPokemonCenterSign, playerDirection);
+        return TRUE;
+    case MB_POKEMART_SIGN:
+        SetUpWalkIntoSignScript(Common_EventScript_ShowPokemartSign, playerDirection);
+        return TRUE;
+    case MB_SIGNPOST:
+        script = GetSignpostScriptAtMapPosition(position);
+        if (script == NULL)
+            return FALSE;
+        SetUpWalkIntoSignScript(script, playerDirection);
+        return TRUE;
+    default:
+        return FALSE;
+    }
+}
+
+static u32 GetFacingSignpostType(u16 metatileBehavior, u32 playerDirection)
+{
+    if (MetatileBehavior_IsPokemonCenterSign(metatileBehavior) == TRUE)
+        return MB_POKEMON_CENTER_SIGN;
+    if (MetatileBehavior_IsPokeMartSign(metatileBehavior) == TRUE)
+        return MB_POKEMART_SIGN;
+    if (MetatileBehavior_IsSignpost(metatileBehavior) == TRUE)
+        return MB_SIGNPOST;
+
+    return NOT_SIGNPOST;
+}
+
+static void SetMsgSignPostAndVarFacing(u32 playerDirection)
+{
+    gWalkAwayFromSignpostTimer = WALK_AWAY_SIGNPOST_FRAMES;
+    gMsgBoxIsCancelable = TRUE;
+	gMsgIsSignPost = TRUE;
+    gSpecialVar_Facing = playerDirection;
+}
+
+static void SetUpWalkIntoSignScript(const u8 *script, u32 playerDirection)
+{
+    ScriptContext_SetupScript(script);
+    SetMsgSignPostAndVarFacing(playerDirection);
+}
+
+static const u8 *GetSignpostScriptAtMapPosition(struct MapPosition *position)
+{
+    const struct BgEvent *event = GetBackgroundEventAtPosition(&gMapHeader, position->x - 7, position->y - 7, position->elevation);
+    if (event == NULL)
+        return NULL;
+    if (event->bgUnion.script != NULL)
+        return event->bgUnion.script;
+    return EventScript_TestSignpostMsg;
+}
+
+static void Task_OpenStartMenu(u8 taskId)
+{
+    if (ArePlayerFieldControlsLocked())
+        return;
+
+    PlaySE(SE_WIN_OPEN);
+    ShowStartMenu();
+    DestroyTask(taskId);
+}
+
+bool32 IsDpadPushedToTurnOrMovePlayer(struct FieldInput *input)
+{
+    return (input->dpadDirection != 0 && GetPlayerFacingDirection() != input->dpadDirection);
+}
+
+void CancelSignPostMessageBox(struct FieldInput *input)
+{
+    if (!ScriptContext_IsEnabled())
+        return;
+
+    if (gWalkAwayFromSignpostTimer)
+    {
+        gWalkAwayFromSignpostTimer--;
+        return;
+    }
+
+	if (!gMsgBoxIsCancelable)
+		return;
```

Comment:
spaces instead of tabs

Reply:
[`f0bd3c5` (#5044)](https://github.com/rh-hideout/pokeemerald-expansion/pull/5044/commits/f0bd3c57daa2a32eca39e01b8f10c92d85d8e4bf)